src/modules/keyboardmanager/ui/EditKeyboardWindow.cpp

```diff
@@ -17,6 +23,88 @@ std::mutex editKeyboardWindowMutex;
 // Stores a pointer to the Xaml Bridge object so that it can be accessed from the window procedure
 static XamlBridge* xamlBridgePtr = nullptr;
 
+static std::vector<DWORD> GetOrphanedKeys()
+{
+    std::set<DWORD> ogKeys;
+    std::set<DWORD> newKeys;
+
+    for (int i = 0; i < SingleKeyRemapControl::singleKeyRemapBuffer.size(); i++)
+    {
+        DWORD ogKey = SingleKeyRemapControl::singleKeyRemapBuffer[i][0];
+        DWORD newKey = SingleKeyRemapControl::singleKeyRemapBuffer[i][1];
+        if (ogKey != 0 && newKey != 0)
+        {
+            ogKeys.insert(ogKey);
+            newKeys.insert(newKey);
+        }
+    }
+
+    for (auto& k : newKeys)
+    {
+        ogKeys.erase(k);
+    }
+
+    return std::vector(ogKeys.begin(), ogKeys.end());
+}
+
+static IAsyncOperation<bool> OrphanKeysConfirmationDialog(
+    KeyboardManagerState& state,
+    const std::vector<DWORD>& keys,
+    XamlRoot root)
+{
+    ContentDialog confirmationDialog;
+    confirmationDialog.XamlRoot(root);
+    confirmationDialog.Title(box_value(L"The following keys are unassigned and you won't be able to use them:"));
+    confirmationDialog.Content(nullptr);
+    confirmationDialog.IsPrimaryButtonEnabled(true);
+    confirmationDialog.DefaultButton(ContentDialogButton::Primary);
+    confirmationDialog.PrimaryButtonText(winrt::hstring(L"Continue Anyway"));
+    confirmationDialog.IsSecondaryButtonEnabled(true);
+    confirmationDialog.SecondaryButtonText(winrt::hstring(L"Cancel"));
+
+    TextBlock orphanKeysBlock;
+    std::wstring orphanKeyString;
+    for (auto k : keys)
+    {
+        orphanKeyString.append(state.keyboardMap.GetKeyName(k));
+        orphanKeyString.append(L", ");
+    }
+    orphanKeyString = orphanKeyString.substr(0, max(0, orphanKeyString.length() - 2));
+    orphanKeysBlock.Text(winrt::hstring(orphanKeyString));
```

Comment:
Can we add this here
`orphanKeysBlock.TextWrapping(TextWrapping::Wrap);`
It will prevent overflowing text if it exceeds the line length (if the user orphans many keys. I was able to hit it by orphaning all the "Browser" keys).